### PR TITLE
Do not lint when range is completely included into another one

### PR DESCRIPTION
**Pull request by ThibsG**
This fix has been developed following this [comment](https://github.com/rust-lang/rust-clippy/issues/5986#issuecomment-703313548).
So this will be linted:
```
|----------|
        |-----------|
```
Now this won't be linted:
```
              |---|
|--------------------|
```
and this will still lint:
```
|--------|
|--------------|
```

Fixes: #5986 

changelog: Fix FPs in match_overlapping_arm, when first arm is completely included in second arm

### CODE REVIEW

File: tests/ui/match_overlapping_arm.rs [hunk 1]
```diff
@@ -57,6 +57,18 @@ fn overlapping() {
...
+        5..7 => println!("5 .. 7"),
+        0..10 => println!("0 .. 10"),
+        _ => (),
+    }
+
+    match 42 {
+        5..10 => println!("5 .. 10"),
+        0..=10 => println!("0 ... 10"),
+        _ => (),
+    }
```
Comment (flip1995) [inline tests/ui/match_overlapping_arm.rs:70]: Please also add tests for the other two cases, that you mentioned in the PR.

Also can you add a test for 

```
|-----------|    (arm 1)
     |----|      (arm 2)
```

which shouldn't be linted by Clippy, but by rustc.

**Review by flip1995 — COMMENTED** [review]
LGTM. Some more testing would be nice, though.

File: tests/ui/match_overlapping_arm.rs [hunk 2]
```diff
@@ -69,6 +69,24 @@ fn overlapping() {
         _ => (),
     }
 
+    match 42 {
+        0..14 => println!("0 .. 14"),
+        5..10 => println!("5 .. 10"),
+        _ => (),
+    }
```
Comment (flip1995) [inline tests/ui/match_overlapping_arm.rs:76]: Huh, shouldn't rustc trigger here? the second arm is unreachable. Well, at least this lint doesn't trigger here. 

A quick check on [Playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=e71febb8617c05922852c5c1e153fe6f) shows that rustc triggers here correctly. So I assume this is a late pass lint in rustc and therefore would only trigger once the clippy lint doesn't trigger anymore.

**Review by flip1995 — APPROVED** [review]

**Comment by flip1995** [comment]
@bors r+

Thanks!

File: tests/ui/match_overlapping_arm.rs (hunk 2 above)
Comment (ThibsG) [reply tests/ui/match_overlapping_arm.rs:76]: Yes I was a bit surprise too, I was also waiting for an unreachable warning :sweat_smile:

**Pull request merged**